Cubical/Data/List/FinData.agda

```diff
@@ -34,12 +64,269 @@ lookup-tabulate : ∀ n → (^a : Fin n → A)
+                → l ↔ (permute l e)
+F≃ (↔permute l e) = invEquiv e ∙ₑ ≡→Fin≃ (sym (length-permute l e))
+l≡ (↔permute l e) k =
+        cong (lookup l) (sym (secEq e k))
+      ∙ sym (lookup-tabulateT _ (lookup l ∘ (fst e)) (invEq e k))
+
+
+isRefl↔ : BinaryRelation.isRefl (_↔_ {A = A})
+F≃ (isRefl↔ a) = idEquiv _
+l≡ (isRefl↔ a) _ = refl
+
+isSym↔ : BinaryRelation.isSym (_↔_ {A = A})
+F≃ (isSym↔ _ b (prm e p)) = invEquiv e
+l≡ (isSym↔ _ b (prm e p)) k = cong (lookup b) (sym (secEq e _)) ∙ sym (p _)
+
+sym↔ : {a b : List A} → a ↔ b → b ↔ a
+sym↔ = isSym↔ _ _
+
+isTrans↔ : BinaryRelation.isTrans (_↔_ {A = A})
+F≃ (isTrans↔ a b c (prm e p) (prm f q)) = e ∙ₑ f
+l≡ (isTrans↔ a b c (prm e p) (prm f q)) _ = p _ ∙ q _
+
+
+-- Some helpful notation:
+
+_∙↔_ : {a b c : List A} → a ↔ b → b ↔ c → a ↔ c
+_∙↔_ {a = a} {b} {c} = isTrans↔ a b c
+
+infixr 30 _∙↔_
+
+_↔⟨_⟩_ : ∀ a → {b c : List A} → a ↔ b → b ↔ c → a ↔ c
+_↔⟨_⟩_ a {b} {c} = isTrans↔ a b c
+
+_■↔ : (a : List A) → (a ↔ a)
+_■↔ = isRefl↔
+
+infixr  0 _↔⟨_⟩_
+infix   1 _■↔
+
+
+isEquivRel↔ : BinaryRelation.isEquivRel {ℓ = ℓ} {A = List A} _↔_
+isEquivRel↔ = BinaryRelation.equivRel isRefl↔ isSym↔ isTrans↔
+
+↔→length≡ : ∀ {x y : List A} →  x ↔ y → length x ≡ length y
+↔→length≡ = isInjectiveFin≃ ∘ F≃
+
+¬nil↔cons : ∀ {x : A} {xs} → ¬ ([] ↔ x ∷ xs)
+¬nil↔cons = ℕznots ∘ ↔→length≡ {x = []}
+
+¬cons↔nil : ∀ {x : A} {xs} → ¬ (x ∷ xs ↔ [])
+¬cons↔nil = ℕsnotz ∘ ↔→length≡ {y = []}
+
+_∷↔_ : ∀ (a : A) {xs ys} → xs ↔ ys → a ∷ xs ↔ a ∷ ys
+a ∷↔ (prm e p) = prm (sucPerm e)  λ { zero → refl ; (suc _) → p _}
+
+≡→↔ : ∀ {xs ys : List A} → xs ≡ ys → xs ↔ ys
+≡→↔ {xs = xs} = J (λ ys p → xs ↔ ys) (isRefl↔ xs)
+
+headSwap↔ : {x y : A} → ∀ {l} → x ∷ y ∷ l ↔ y ∷ x ∷ l
+headSwap↔ =
+  prm swap0and1≃ λ { zero → refl ; one → refl ; (suc (suc k)) → refl }
+
+
+_∷↔∷ʳ_ : ∀ xs → (x : A) → xs ∷ʳ x ↔ x ∷ xs
+_∷↔∷ʳ_ =
+  ind (isRefl↔ ∘ [_])
+        λ x _ → ≡→↔ (++-assoc [ _ ] _ [ _ ])
+          ∙↔ (_ ∷↔ x _)
+          ∙↔ headSwap↔
+
+_↔∷ʳ_ : ∀ {xs ys} → xs ↔ ys → ∀ (a : A) → xs ∷ʳ a ↔ ys ∷ʳ a
+r ↔∷ʳ _ = (_ ∷↔∷ʳ _) ∙↔ (_ ∷↔ r) ∙↔ sym↔ (_ ∷↔∷ʳ _)
+
+
+_++↔_ : (x y : List A) → x ++ y ↔ y ++ x
+x ++↔ [] = ≡→↔ (++-unit-r x)
+[] ++↔ y@(_ ∷ _) = ≡→↔ (sym (++-unit-r y) )
+(x ∷ xs) ++↔ (y ∷ ys) =
+       (x ∷↔ ((xs ++↔ (y ∷ ys))))
+    ∙↔ headSwap↔
+    ∙↔ (y ∷↔ (cong↔++R (sym↔ (ys ∷↔∷ʳ x)) xs)
+              ∙↔ (≡→↔ (++-assoc ys [ x ] xs)))
+  where
+
+  lookup-FinSumChar : ∀ {xs ys : List A} →
+          ∀ k → lookup (xs ++ ys) k ≡
+           ⊎.rec (lookup xs) (lookup ys)
+             (FinSumChar.inv (length xs) (length ys)
+               (subst Fin (length++ xs ys) k))
+  lookup-FinSumChar {xs = []} {ys} _ = cong (lookup ys) (sym (transportRefl _))
+  lookup-FinSumChar {xs = x ∷ xs} {ys} zero =
+     cong (⊎.rec (lookup (x ∷ xs)) (lookup ys) ∘ (FinSumChar.inv _ _))
+       (transportFin-zero _)
+  lookup-FinSumChar {xs = x ∷ xs} {ys} (suc _) =
+     _ ≡⟨ lookup-FinSumChar {xs = xs} _ ⟩
```

Comment:
```≡⟨ ... ⟩``` is supposed to be used, if you want to write down the endpoints of the equalities. If you don't want to do that, just use the concatenation operator ```\.```. But I think here it would actually help to fill in those ```_```